crates/store/re_dataframe2/src/query.rs

```diff
@@ -726,3 +739,681 @@ impl<'a> QueryHandle<'a> {
         std::iter::from_fn(move || self.next_row_batch())
     }
 }
+
+// ---
+
+#[cfg(test)]
+#[allow(clippy::iter_on_single_items)]
+mod tests {
+    use std::sync::Arc;
+
+    use re_chunk::{Chunk, ChunkId, RowId, TimePoint};
+    use re_chunk_store::{ChunkStore, ChunkStoreConfig, ResolvedTimeRange, TimeInt};
+    use re_log_types::{
+        build_frame_nr, build_log_time,
+        example_components::{MyColor, MyLabel, MyPoint},
+        EntityPath, Timeline,
+    };
+    use re_types_core::Loggable as _;
+
+    use crate::QueryCache;
+
+    use super::*;
+
+    // TODO(cmc): at least one basic test for every feature in `QueryExpression2`.
+    // In no particular order:
+    // * [x] filtered_index
+    // * [x] filtered_index_range
+    // * [x] view_contents
+    // * [x] selection
+    // * [ ] filtered_index_values
+    // * [ ] sampled_index_values
+    // * [ ] filtered_point_of_view
+    // * [ ] sparse_fill_strategy
+
```

Comment:
One area that we might need to semantically define and test is around timeline column. When does a given timeline show up in the result (assuming `selection = None`)?

1) If the timeline exists at all in the store?
2) If anything at all is logged on it in the context of the view content?
3) If it is not empty in the query results?